src/components/calendar/DayInfo.jsx

```diff
@@ -4,23 +4,38 @@ import PropTypes from 'prop-types';
 import { useStoreState } from 'easy-peasy';
 import Text from '@/components/shared/Text';
 import { Ionicons } from '@expo/vector-icons';
+import { isPeak, isFertile, isInfertile } from '@/utils/conclusionProcessing';
 import tailwind, { getColor } from '@/utils/tailwind';
 import DateHeader from '@/components/shared/DateHeader';
+import DayInfoConclusion from './DayInfoConclusion';
 import DayInfoNoReport from './DayInfoNoReport';
 import DayInfoRelation from './DayInfoRelation';
 
 export default function DayInfo({ openReport, openRelationReport }) {
   const day = useStoreState((state) => state.calendar.selectedDay);
-  const conclusions = useStoreState((state) => state.calendar.conclusions[day.key]);
   const propertiesReport = useStoreState((state) => state.calendar.propertiesReports[day.key]);
   const relationReport = useStoreState((state) => state.calendar.relationReports[day.key]);
-  console.log(relationReport);
+
+  function calculateConclusionType() {
+    const conclusion = useStoreState((state) => state.calendar.conclusions[day.key]);
+    if (conclusion) {
+      if (isPeak(conclusion)) return 'peak';
+      if (isFertile(conclusion)) return 'fertile';
+      if (isInfertile(conclusion)) return 'infertile';
+    }
+    return null;
+  }
+
+  const conclusionType = calculateConclusionType();
```

Comment:
Es medio curioso este caso, porque a primera vista, no se está cumpliendo la [primera regla de los hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level), pero en verdad si se cumple, dado que llamas a una función justo abajo, evitando que llamado al hook sea condicional o en otro orden respecto al resto (que es efectivamente el problema de _nestearlos_)

Por legibilidad y pensando en devs que no se manejen tanto en React, lo dejaría fuera e incluso usaría un `useMemo` para no nombrar a la función:

```suggestion
  const conclusion = useStoreState((state) => state.calendar.conclusions[day.key]);

  const conclusionType = useMemo(() => {
    if (conclusion) {
      if (isPeak(conclusion)) return 'peak';
      if (isFertile(conclusion)) return 'fertile';
      if (isInfertile(conclusion)) return 'infertile';
    }
    return null;
  }, [conclusion]);
```

Lo del `useMemo` no es necesario (generalmente lo prefiero porque encuentro que declarar + llamar a una función es más verboso que directamente declararla en el memo), pero si o si movería el `useStoreState`

Reply:
Buena!! voy a cambiarlo por el `useMemo`! 😄